spec/unit/provider/registry_key_spec.rb

```diff
@@ -289,6 +297,8 @@
     let(:testval1_wrong_type) { { :name => "one", :type => :string, :data => 255.chr * 1 } }
     let(:testval1_wrong_data) { { :name => "one", :type => :binary, :data => 254.chr * 1 } }
     let(:testval2) { { :name => "two", :type => :binary, :data => 0.chr * 1 } }
+    let(:testval3) { { :name => "three", :type => :binary, :data => 0.chr * 1 } }
+    let(:testval3_data_string) { { :name => "three", :type => :binary, :data => 0.chr * 1 } }
```

Comment:
Let's refactor these tests. This compromise means the test structure has us backed into a corner.

I'll try to get some time for thoughts on that later today.